core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -1400,4 +1401,20 @@ private static <T> T getValue(Function<FlagsProvider, @Nullable T> method,
     }
 
     private Flags() {}
+
+    // These static variables are defined at the end of this file deliberately
+    // to ensure that all static variables beforehand are initialized.
+    private static final boolean INITIALIZED;
+
+    static {
+        INITIALIZED = true;
+    }
+
+    /**
+     * Returns whether this class has initialized all relevant flags.
+     */
+    @UnstableApi
+    public static boolean initialized() {
+        return INITIALIZED;
+    }
```

Comment:
How about hiding this from the public API unless it's supposed to be used by a user? We could move `INITIALIZED` and `initialized()` to an internal class, e.g.

```java
package com.linecorp.armeria.internal.common;
public final class FlagsLoaded {
    private static boolean loaded;

    public static boolean get() {
        return loaded;
    }

    public static boolean set() {
        loaded = true;
    }
}

package com.linecorp.armeria.common;

public final class Flags {
    ...

    static {
        FlagsLoaded.set();
    }
    ...
}

// in appender
if (!FlagsLoaded.get()) {
    ...
}
```

Reply:
Good point! 👍